policy/scan/local_scanner.go

```diff
@@ -514,7 +514,8 @@ func (s *LocalScanner) RunAssetJob(job *AssetJob) {
 	job.Reporter.AddReport(job.Asset, results)
 
 	upstream := s.upstreamServices(job.Ctx, job.UpstreamConfig)
-	if upstream != nil {
+	// The vuln report is relevant only when we have an aggregate reporter
+	if _, isAggregateReporter := job.Reporter.(*AggregateReporter); upstream != nil && isAggregateReporter {
```

Comment:
it might be cleaner to remove `AddVulnReport` from the Reporter interface and have it be part of an extension interface. That way, is someone decides to make another reported and wants it, things keep working

Reply:
good call! Updated the PR

Reply:
I think you would want to remove the implementations from `reporter_error.go` and `reporter_noop.go`